packages/service-library/src/servicelib/deferred_tasks/_task_schedule.py

```diff
@@ -23,37 +25,64 @@ class TaskState(str, Enum):
 
 
 class TaskScheduleModel(BaseModel):
-    timeout: timedelta = Field(
-        ..., description="Amount of time after which the task execution will time out"
-    )
-    class_unique_reference: ClassUniqueReference = Field(
-        ...,
-        description="reference to the class containing the code and handlers for the execution of the task",
-    )
-    start_context: StartContext = Field(
-        ...,
-        description="data used to assemble the ``StartContext``",
-    )
-
-    state: TaskState = Field(
-        ..., description="represents the execution step of the task"
-    )
-
-    execution_attempts: NonNegativeInt = Field(
-        ...,
-        description="remaining attempts to run the code, only retries if this is > 0",
-    )
-
-    time_started: datetime = Field(
-        default_factory=lambda: arrow.utcnow().datetime,
-        description="time when task schedule was created, used for statistics",
-    )
-
-    result: TaskExecutionResult | None = Field(
-        default=None,
-        description=(
-            f"Populated by {TaskState.WORKER}. It always has a value after worker handles it."
-            "Will be used "
+    timeout: Annotated[
+        timedelta,
+        Field(
+            description="Amount of time after which the task execution will time out"
         ),
-        discriminator="result_type",
-    )
+    ]
+    class_unique_reference: Annotated[
+        ClassUniqueReference,
+        Field(
+            description="reference to the class containing the code and handlers for the execution of the task",
+        ),
+    ]
+    start_context: Annotated[
+        StartContext,
+        Field(
+            description="data used to assemble the ``StartContext``",
+        ),
+    ]
+
+    state: Annotated[
+        TaskState, Field(description="represents the execution step of the task")
+    ]
+
+    total_attempts: Annotated[
+        NonNegativeInt,
+        Field(
+            description="maximum number of attempts before giving up (0 means no retries)"
+        ),
+    ]
+
+    execution_attempts: Annotated[
+        NonNegativeInt,
+        Field(
+            description="remaining attempts to run the code, only retries if this is > 0",
+        ),
+    ]
+
+    wait_cancellation_until: Annotated[
+        datetime | None,
+        Field(description="when set has to wait till this before cancelling the task"),
+    ] = None
+
+    time_started: Annotated[
+        datetime,
+        Field(
+            default_factory=lambda: arrow.utcnow().datetime,
+            description="time when task schedule was created, used for statistics",
+        ),
+    ] = DEFAULT_FACTORY
+
+    result: Annotated[
+        TaskExecutionResult | None,
+        Field(
+            default=None,
```

Comment:
you should not use dfault here if you are using Annotated. Only default_factory makes sense in this scenario